### ISSUE TITLE
Write Back to Firmware

### ISSUE THREAD

**Issue by UberMicro** [Ryochan7/DS4Windows#1749]
I repair controllers Xbox and PlayStation. I buy the original ALPS analog stick modules and even with brand new ones the center is off and have to manually calibrate them. This add significant time to the repair. Specially no I'm getting a lot of work in. 

Is there anyway once a new module has been replaced to tell the firmware of the controller the new 0 (zero, centre). I assume these are calibrated at factory as there is deviation in the potentiometers. This is why even if you have two perfectly new and centering controllers and you swap an analog module over there is drift the stick wont be centered. Yes I have done this, and just done it to a new PS5 controller.

Basically after a new module is soldered in X axis would read +14 and Y axis could read -7 . I'm sure at the factory they would plug in the controllers to a machine to load FW and calibrate all the sticks to what ever value they are at, it would write it as 0 for both X and Y. 

I have heard rumors from some Chinese suppliers that there is a "DS4 Correction Tool" (PS4 Correction Tool) that does this but no-one is letting on where it can be found. 

Schematic for Xbox Elite Controller would be nice as well!

**Comment by mika-n** [comment]
Maybe but maybe not an easy task. And as far I know Sony itself has not published publicly available apps and tools to update genuine DS4 pads. Some copy-cat fake controllers have FW update capability.
https://securelist.com/hacking-microcontroller-firmware-through-a-usb/89919/
